src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Kafka/KafkaConsumerConsumeIntegration.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka
+{
+    /// <summary>
+    /// Confluent.Kafka Consumer.Consume calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "Confluent.Kafka",
+        TypeName = "Confluent.Kafka.Consumer`2",
+        MethodName = "Consume",
+        ReturnTypeName = KafkaConstants.ConsumeResultTypeName,
+        ParameterTypeNames = new[] { ClrNames.Int32 },
+        MinimumVersion = "1.4.0",
+        MaximumVersion = "1.*.*",
+        IntegrationName = KafkaConstants.IntegrationName)]
+    public class KafkaConsumerConsumeIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="millisecondsTimeout">The maximum period of time the call may block.</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, int millisecondsTimeout)
+        {
+            // If we are already in a consumer scope, close it, and start a new one on method exit.
+            KafkaHelper.CloseConsumerScope(Tracer.Instance);
+            return CallTargetState.GetDefault();
```

Comment:
If `Tracer.Instance.Settings.KafkaCreateConsumerScopeEnabled` is set and the span only represents the consume action, I'm thinking you could get the current DateTimeOffset here so the duration of the span represents when the method started and returned. You can see one example in the RabbitMQ integration for reference below. Do you think this would be useful?

https://github.com/DataDog/dd-trace-dotnet/blob/33ca4afc9c935b9800a12d934821430af8a2d0e2/src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/RabbitMQ/BasicGetIntegration.cs#L36-L39

Reply:
If I'm understanding correctly, I'm not sure if it's very useful in the Kafka case, as it would often be a "meaningless" number IMO, given `Consumer.Consume()` can block for an arbitrary amount of time. 

For example, imagine my message handler takes 100ms to run, but a message isn't available for 3s. Using this approach would give the impression that the message-handling loop took 3.1s, instead of the 0.1s it would with the current implementation?

Reply:
Got it, I forgot that the `Consumer.Consume()` was blocking. The current implementation sounds good then!